### DISCUSSION TITLE
Dynamic imports in Webpack

### DISCUSSION THREAD

**Discussion by imageck** [webpack/webpack#17610] [Q&A]
In a Next.js project, I've noticed that `import()` calls are transformed into `__webpack_require__`. Where does it come from? Is it a Babel or SWC transform, or does Webpack always do it? I'm curious how it's implemented and what kind of errors it may throw.

Could you please point me to the source code?

**Comment by manyuemeiquqi** [comment]
It's a implement of webpack to translate cjs esm module to webpack runtime. The primary reason for this transformation is that the <script> tag in HTML does not provide built-in support for module loading without `type:module`. When a JavaScript file is included using a <script> tag, it is treated as a global script, and the code within it is executed immediately in the global scope. If you interested in it, you can implement mini webpack to learn more.
